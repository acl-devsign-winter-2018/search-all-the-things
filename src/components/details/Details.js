import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


export default class Details extends Component {
    
  static propTypes = {
    searchDetail: PropTypes.object.isRequired
  };


  render() {
    const { searchDetail } = this.props;

    if(searchDetail === null) return null;

    return (
      <div>
        <Link to="/search">Search</Link>
        <pre>{JSON.stringify(searchDetail, true, 2)}</pre>
      </div>
    );
  }
}