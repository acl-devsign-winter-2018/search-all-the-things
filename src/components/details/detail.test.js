import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Details from './Details';

describe ('Details', () => {
  it('renders search details', () => {
    const search = {   
      'name': 'Luke Skywalker',
      'height': '172',
      'mass': '77',
      'hair_color': 'blond',
      'skin_color': 'fair',
      'eye_color': 'blue',
      'birth_year': '19BBY',
      'gender': 'male',
    };

    const wrapper = shallow(<Details details={search}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();

  });
});