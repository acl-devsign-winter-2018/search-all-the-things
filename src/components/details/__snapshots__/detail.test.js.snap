// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders search details 1`] = `
<div>
  <Link
    replace={false}
    to="/search"
  >
    Search
  </Link>
  <pre />
</div>
`;
